File: abc135/abc135_c.cs
// https://atcoder.jp/contests/abc135/submissions/8233164

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var A = Console.ReadLine().Split(' ').Select(long.Parse).ToArray();
            var B = Console.ReadLine().Split(' ').Select(long.Parse).ToArray();

            var result = 0L;
            for (int i = 0; i < N; i++)
            {
                if (B[i] > A[i])
                {
                    B[i] -= A[i];
                    result += A[i];
                    A[i] = 0;
                }
                else
                {
                    A[i] -= B[i];
                    result += B[i];
                    B[i] = 0;
                }

                if (B[i] > A[i + 1])
                {
                    B[i] -= A[i + 1];
                    result += A[i + 1];
                    A[i + 1] = 0;
                }
                else
                {
                    A[i + 1] -= B[i];
                    result += B[i];
                    B[i] = 0;
                }
            }
            Console.WriteLine(result);
        }
    }
}

File: abc/abc135/abc135_c.fs
// https://atcoder.jp/contests/abc135/submissions/11958270

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int64
let B = stdin.ReadLine().Split() |> Array.map int64

let rec resolver s n =
    if n = N then s
    else
        let mutable t = 0L
        if B.[n] > A.[n] then
            B.[n] <- B.[n] - A.[n]
            t <- t + A.[n]
            A.[n] <- 0L
        else
            A.[n] <- A.[n] - B.[n]
            t <- t + B.[n]
            B.[n] <- 0L
        if B.[n] > A.[n + 1] then
            B.[n] <- B.[n] - A.[n + 1]
            t <- t + A.[n + 1]
            A.[n + 1] <- 0L
        else
            A.[n + 1] <- A.[n + 1] - B.[n]
            t <- t + B.[n]
            B.[n] <- 0L
        resolver (s + t) (n + 1)

resolver 0L 0 |> stdout.WriteLine
